Google csp

jonverson opened the pull request:
[ ] Check if this is a typo or other quick fix and ignore the rest :)

## Type of change
Updated the CSP settings. Currently the edu site is failing to load many references to connect-src, script-src, img-src and frame-src. I added all the scripts that were failing on the edu site.

### What should this PR do?
This PR should resolve current broken CSP console errors.

### Why are we making this change?
To allow tag manger to track user visits to edu site and allow scripts on the site to work correctly.

### What are the acceptance criteria? 
The CSP console errors should be cleared up when loading the edu site

### How should this PR be tested?
You can pull this branch down locally test if there are any CSP errors in the console. If there is a staging site this should also work there as well.

jonverson committed:
updated csp with missing refernces to www.google.com

jonverson committed:
added more csp entries to additonal files

jonverson committed:
added more CSP exceptions for scripts to load

jonverson committed:
added another script to tje CSP list

ltagliaferri reviewed (commented):
It looks like there is more than just Google Tag Manager links being added here... Would you be able to walk me through them and what they are used for? How can we access the data from these?

We removed the 6sense one based on the understanding that we were sunsetting that.

jonverson commented:
GTM is just the vehicle for loading all the scripts. The scripts that I
added are all ones that chrome was blocking and complaining about. If these
scripts aren't white listed the edu site will block them and they won't
load.

On Mon, Sep 29, 2025 at 6:51 PM ltagliaferri ***@***.***>
wrote:

> ***@***.**** commented on this pull request.
>
> It looks like there is more than just Google Tag Manager links being added
> here... Would you be able to walk me through them and what they are used
> for? How can we access the data from these?
>
> We removed the 6sense one based on the understanding that we were
> sunsetting that.
> ------------------------------
>
> In layouts/partials/head/script-header.html
> <https://github.com/chainguard-dev/edu/pull/2634#discussion_r2389579336>:
>
> >         default-src 'self';
> -       frame-src 'self' edu.chainguard.dev https://player.vimeo.com https://www.youtube.com https://www.youtube-nocookie.com https://platform.twitter.com https://syndication.twitter.com https://visualization-ui.chainguard.app https://terminal.inky.wtf;
> +       frame-src 'self' edu.chainguard.dev https://www.googletagmanager.com https://player.vimeo.com https://www.youtube.com https://www.youtube-nocookie.com https://platform.twitter.com https://syndication.twitter.com https://visualization-ui.chainguard.app https://terminal.inky.wt https://app.qualified.com/;
>
> ⬇️ Suggested change
>
> -       frame-src 'self' edu.chainguard.dev https://www.googletagmanager.com https://player.vimeo.com https://www.youtube.com https://www.youtube-nocookie.com https://platform.twitter.com https://syndication.twitter.com https://visualization-ui.chainguard.app https://terminal.inky.wt https://app.qualified.com/;
> +       frame-src 'self' edu.chainguard.dev https://www.googletagmanager.com https://player.vimeo.com https://www.youtube.com https://www.youtube-nocookie.com https://platform.twitter.com https://syndication.twitter.com https://visualization-ui.chainguard.app https://terminal.inky.wtf https://app.qualified.com/;
>
> ------------------------------
>
> In layouts/partials/head/script-header.html
> <https://github.com/chainguard-dev/edu/pull/2634#discussion_r2389580013>:
>
> >         style-src 'self' 'unsafe-inline' edu.chainguard.dev https://tagmanager.google.com https://www.googletagmanager.com https://googletagmanager.com cdn.jsdelivr.net https://fonts.googleapis.com https://unpkg.com https://use.fontawesome.com;
>         form-action 'self';
>         font-src 'self' edu.chainguard.dev https://fonts.googleapis.com https://fonts.gstatic.com https://cdn.jsdelivr.net https://use.fontawesome.com;
> -       script-src 'self' 'unsafe-inline' 'unsafe-eval' edu.chainguard.dev https://www.googletagmanager.com https://tagmanager.google.com https://googletagmanager.com *.googleapis.com cdn.jsdelivr.net *.googletagmanager.com https://www.google-analytics.com https://analytics.google.com https://snap.licdn.com https://unpkg.com https://csp.withgoogle.com https://play.google.com http://localhost:1313 http://localhost:8080;
> -       connect-src 'self' https://www.googletagmanager.com https://tagmanager.google.com https://googletagmanager.com *.google-analytics.com *.googletagmanager.com https://analytics.google.com https://region1.google-analytics.com https://www.google-analytics.com https://px.ads.linkedin.com https://px4.ads.linkedin.com https://storage.googleapis.com https://packages.wolfi.dev https://script.google.com https://script.googleusercontent.com https://csp.withgoogle.com https://play.google.com *.googleapis.com;
> -       img-src 'self' 'unsafe-inline' https://ssl.gstatic.com edu.chainguard.dev https://storage.googleapis.com https://www.googletagmanager.com https://googletagmanager.com https://www.google-analytics.com https://analytics.google.com https://px.ads.linkedin.com https://px4.ads.linkedin.com data:;
> +       script-src 'self' 'unsafe-inline' 'unsafe-eval' edu.chainguard.dev https://www.google.com https://www.googletagmanager.com https://tagmanager.google.com https://googletagmanager.com *.googleapis.com cdn.jsdelivr.net *.googletagmanager.com https://www.google-analytics.com https://analytics.google.com https://snap.licdn.com https://unpkg.com https://csp.withgoogle.com https://play.google.com http://localhost:1313 http://localhost:8080 https://www.redditstatic.com http://js.hs-scripts.com https://tag.clearbitscripts.com https://j.6sc.co https://tracking.g2crowd.com https://tag.unifyintent.com https://static.reo.dev https://googleads.g.doubleclick.net https://js.qualified.com https://js.hscollectedforms.net https://js.hubspot.com https://js.hsadspixel.net https://js.hs-banner.com https://js.hs-analytics.net https://x.clearbitjs.com https://js.zi-scripts.com;
>
> Are all of these for Google Tag Manager?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/chainguard-dev/edu/pull/2634#pullrequestreview-3282245692>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAHUVK733IVZJSEUIYKC2X33VHH2JAVCNFSM6AAAAACH23MOUGVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZTEOBSGI2DKNRZGI>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>
